### PR TITLE
Cleanup issues in ReelMagic flagged by Coverity

### PR DESCRIPTION
**Pull request by kcgen**
Also:
 - always registers the ReelMagic messages (for translation)
 - renames the 'default' key to `common` to disambiguate it with the description for `auto`
 - fixes inability to set `reelmagic_key = thehorde`

Suggest reviewing commit-by-commit.
(Note that coding style is meant to fit w/ the RealMagic sources).

### CODE REVIEW

File: src/hardware/reelmagic/driver.cpp [hunk 1]
```diff
@@ -1367,6 +1380,11 @@ void ReelMagic_Init(Section* sec)
 	assert(sec);
 	const auto section = static_cast<Section_prop*>(sec);
 
+	// Always register the driver's text messages, even if the card is
+	// disabled We cannot rely on the driver to register them because we
+	// only create the file if the user wants ReelMagic enabled.
+	FMPDRV::AddMessages();
```
Comment (FeralChild64) [inline src/hardware/reelmagic/driver.cpp:1386]: I would, in my opinion, be more sane to execute this from within `REELMAGIC_MaybeCreateFmpdrvExecutable` - to ensure that string order in the translation file is the same regardless of the configuration.

**Comment by FeralChild64** [comment]
@kcgen I've noticed a small problem - if ReelMagic is emulated (`Z:\FMPDRV.EXE` is present), the output of command `HELP` does not fit one one screen any more. I would expect `HELP` to fit in one screen and `HELP /a` to list all the commands, spanning across as many screens as needed.

File: src/hardware/reelmagic/driver.cpp (hunk 1 above)
Comment (kcgen) [reply src/hardware/reelmagic/driver.cpp:1386]: Yes: only one place to add messages instead of two, that's even better, @FeralChild64 - thanks.

**Comment by kcgen** [comment]
> @kcgen I've noticed a small problem - if ReelMagic is emulated (`Z:\FMPDRV.EXE` is present), the output of command `HELP` does not fit on one screen any more. 

Ok, I've demoted FMPDRV from being a 'Common' application; so now it only shows up in the `/all` listing (which makes sense.. this isn't an everyday tool that people will be using).

Here's the output:

help-without-reelmagic-enabled

---

I would expect `HELP` to fit in one screen and `HELP /a` to list all the commands, spanning across as many screens as needed.

Let's ignore ReelMagic and FMPDRV for now: if we always want  `HELP`'s output to fit on one screen, then it looks like we've hit that limit already.

Programmatically, this means that only 15 CLI tools can be given the `HELP_Filter::Common` designation -- and everything else is demoted to `HELP_Filter::All`. (which is fine).

Actually, maybe we need to cut back evern more, because just `HELP` already scrolls past just by a little bit.  Here's `HELP | MORE`:

help-more-without-reelmagic-enabled

File: src/hardware/reelmagic/mpeg_decoder.h [hunk 2]
```diff
@@ -929,14 +962,18 @@ int plm_init_decoders(plm_t *self) {
 }
 
 void plm_destroy(plm_t *self) {
+	if (!self)
+		return;
```
Comment (johnnovak) [inline src/hardware/reelmagic/mpeg_decoder.h:966]: Braces?

File: src/hardware/reelmagic/mpeg_decoder.h [hunk 3]
```diff
@@ -1392,18 +1429,38 @@ plm_buffer_t *plm_buffer_create_with_filename(const char *filename) {
 }
 
 plm_buffer_t *plm_buffer_create_with_file(FILE *fh, int close_when_done) {
+	if (!fh)
+		return NULL;
```
Comment (johnnovak) [inline src/hardware/reelmagic/mpeg_decoder.h:1433]: Braces?

File: src/hardware/reelmagic/mpeg_decoder.h [hunk 4]
```diff
@@ -1463,6 +1520,9 @@ plm_buffer_t *plm_buffer_create_for_appending(size_t initial_capacity) {
 }
 
 void plm_buffer_destroy(plm_buffer_t *self) {
+	if (!self)
+		return;
```
Comment (johnnovak) [inline src/hardware/reelmagic/mpeg_decoder.h:1524]: Braces?

File: src/hardware/reelmagic/mpeg_decoder.h [hunk 5]
```diff
@@ -1774,12 +1845,14 @@ plm_demux_t *plm_demux_create(plm_buffer_t *buffer, int destroy_when_done) {
...
 	self->start_code = -1;
 
-	plm_demux_has_headers(self);
 	return self;
 }
 
 void plm_demux_destroy(plm_demux_t *self) {
-	if (self->destroy_buffer_when_done) {
+	if (!self)
+		return;
```
Comment (johnnovak) [inline src/hardware/reelmagic/mpeg_decoder.h:1853]: Braces?

File: src/hardware/reelmagic/mpeg_decoder.h [hunk 6]
```diff
@@ -2720,7 +2799,10 @@ plm_video_t * plm_video_create_with_buffer(plm_buffer_t *buffer, int destroy_whe
 }
 
 void plm_video_destroy(plm_video_t *self) {
-	if (self->destroy_buffer_when_done) {
+	if (!self)
+		return;
```
Comment (johnnovak) [inline src/hardware/reelmagic/mpeg_decoder.h:2803]: Braces?

File: src/hardware/reelmagic/mpeg_decoder.h [hunk 7]
```diff
@@ -3846,7 +3934,10 @@ plm_audio_t *plm_audio_create_with_buffer(plm_buffer_t *buffer, int destroy_when
 }
 
 void plm_audio_destroy(plm_audio_t *self) {
-	if (self->destroy_buffer_when_done) {
+	if (!self)
+		return;
```
Comment (johnnovak) [inline src/hardware/reelmagic/mpeg_decoder.h:3938]: Braces?

**Comment by johnnovak** [comment]
Looks good @kcgen, nice explanatory comments & error checking/cleanup code. Would it make sense to extend the clang formatter to enforce braces around single-line statements as well?

**Review by johnnovak — APPROVED** [review]

**Comment by kcgen** [comment]
> clang formatter to enforce braces around single-line 

Added! Didn't realize it could do this, but now see that version 15.x added it.

**Pull request merged**